File: lab6-python/lecture/queue.py
class Queue:
    def __init__(self):
        self._q = []
        self._size = 0

    def empty(self) -> int:
        return self._size == 0
    
    def size(self) -> int:
        return self._size
    
    def front(self) -> str:
        return self._q[0]
    
    def push(self, item : str) -> None:
        self._size += 1
        self._q.append(item)
    
    def pop(self) -> str:
        self._size -= 1
        return self._q.pop(0)    

File: lab7-java/DataStructures/src/examples/Queue.java
package examples;

import java.util.ArrayList;

public class Queue<T> implements Container<T> {
    private ArrayList<T> _container;
    public Queue()
    {
        this._container = new ArrayList<>();
    }

    @Override
    public boolean empty()
    {
        return this._container.isEmpty();
    }

    @Override
    public int size()
    {
        return this._container.size();
    }
    @Override
    public T next() {
        return this._container.getFirst();
    }

    @Override
    public void push(T elt) {
        this._container.add(elt);
    }

    @Override
    public T pop() {
        return this._container.removeFirst();
    }
}
